Firestore/core/src/firebase/firestore/CMakeLists.txt

```diff
@@ -23,6 +23,18 @@ cc_library(
     firebase_firestore_util
 )
 
+cc_library(
+  firebase_firestore_version
+  SOURCES
+    firestore_version.cc
+)
+
+target_compile_definitions(
```

Comment:
The general strategy seems OK, though I'd suggest putting this in `util` so that it's automatically available implicitly.

Reply:
Sorry, do you mean putting this `target_compile_definitions` in `util/CMakeLists.txt`? (I'm asking because it's used from `remote`, not `util`).

Reply:
The `FIR`-named version is a public symbol so the value must be available not just to send in RPC headers.

I don't have any strong preference for where this goes, but functionally it's a part of the Core API that's exposed by the Objective-C API.

If the other Firebase products have a public C++ version symbol, this should look like those and likely be a peer of things like `Timestamp` and `GeoPoint`.

If the other Firebase products don't have such a thing, then this could go anywhere, but I don't think we need to build a separate library to do it. On Android, we log the Firestore version with every log message so it seems reasonable for us to build the version there.

Reply:
Other Firebase C++ products seem to have a generated `version.h` file that lives at the topmost project-specific directory, so following that example, the correct location would be `firebase/firestore`. I'm just not sure in which library to put this as far as CMake goes. `Timestamp` and `GeoPoint` are in `firebase_firestore_types`; having version in a library called "types" seems a little weird (alternatively, I could rename the library from "types" to something more generic). I left it as `firebase_firestore_version` for now, let me know if you have a different preference.

Reply:
This seems fine and we can always refactor later if something more obvious comes to mind as we do more integration work.